rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -0,0 +1,1007 @@
+                application_context.clone(),
+                client.clone(),
+                &aep_name_command_options,
+            )),
+            update_asset_status_command_invoker: Arc::new(UpdateAssetStatusCommandInvoker::new(
+                application_context.clone(),
+                client.clone(),
+                &aep_name_command_options,
+            )),
+            create_detected_asset_command_invoker: Arc::new(
+                CreateDetectedAssetCommandInvoker::new(
+                    application_context.clone(),
+                    client.clone(),
+                    &aep_name_command_options,
+                ),
+            ),
+            notify_on_asset_update_command_invoker: Arc::new(
+                NotifyOnAssetUpdateCommandInvoker::new(
+                    application_context.clone(),
+                    client.clone(),
+                    &aep_name_command_options,
+                ),
+            ),
+            create_asset_endpoint_profile_command_invoker: Arc::new(
+                CreateDiscoveredAssetEndpointProfileCommandInvoker::new(
+                    application_context,
+                    client.clone(),
+                    &AepCommandOptionsBuilder::default()
+                        .topic_token_map(HashMap::from([(
+                            "ex:connectorClientId".to_string(),
+                            client.client_id().to_string(),
+                        )]))
+                        .build()
+                        .expect("Statically generated options should not fail."),
+                ),
+            ),
+            aep_update_event_telemetry_dispatcher,
+            asset_update_event_telemetry_dispatcher,
+        }
+    }
+
+    /// Retrieves an asset endpoint profile from a Azure Device Registry service.
+    ///
+    /// # Arguments
+    /// * `aep_name` - The name of the asset endpoint profile.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns a [`AssetEndpointProfile`] if the the endpoint was found.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidArgument`](ErrorKind::InvalidArgument)
+    /// if the `timeout` is zero or > `u32::max`, or there is an error building the request.
+    ///
+    /// [`struct@Error`] of kind [`ServiceError`](ErrorKind::ServiceError)
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn get_asset_endpoint_profile(
+        &self,
+        aep_name: String,
+        timeout: Duration,
+    ) -> Result<AssetEndpointProfile, Error> {
+        let command_request = rpc_command::invoker::RequestBuilder::default()
+            .topic_tokens(HashMap::from([("ex:aepName".to_string(), aep_name)]))
```

Comment:
same here, I think the `ex` will be added by codegen

Reply:
If these topic token names are used multiple places, it would be good to pull them out into a const